python/lsst/daf/butler/_butler.py

```diff
@@ -1770,3 +1803,69 @@ def prepend_collection_chain(
         transactions short.
         """
         raise NotImplementedError()
+
+    @abstractmethod
+    def extend_collection_chain(
```

Comment:
I also had that thought... but I'm concerned we're going to eventually add options that only apply to some subset of the modes.  Then we'll end up in the situation we're at with a lot of the other Butler methods where it takes multiple pages of documentation/checks to explain which sets of parameters are legal together.

One other idea I had was maybe naming them `collect_chain_*` instead of `*_collection_chain`, so they group together a little better in the docs.  But the rest of the butler methods are verb-first.